src/framework/core/op_list.c

```diff
@@ -175,6 +185,7 @@ int op_list_purge(struct op_list* list)
  */
 void op_list_garbage_collect(struct op_list* list)
 {
+    _op_list_collect_deleted(list);
```

Comment:
This was causing issues with channels_hashtab because some of the channel objects were not getting deleted when op_hashtab_garbage_collect() was called.

The channel shared memory was getting reused by the next garbage collection call causing very strange issues with the different fds from expected getting closed when the channel object was deleted.